File: Data Structures & Algorithms/two-integer-sum/submission-0.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        map1 = {}
        ind = 0
        for i in nums:
            comp = target - i
            if i in map1:
                return [map1[i], ind]
            else:
                map1[comp]=ind
            ind+=1

        

File: Data Structures & Algorithms/duplicate-integer/submission-0.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        std::unordered_map<int,int> m;
        
        for (int i: nums){
            if (m.contains(i)){
                return true;
            }
            m[i] = 1;

        }
        return false;
    }
};